pkg/auth/user/util.go

```diff
@@ -25,3 +30,44 @@ func ConvertAttributes(attrMap map[string][]string) []*v1.UserAttribute {
 	})
 	return result
 }
+
+type loggableAuthProvider struct {
+	Id   string
+	Name string
+	Type string
+}
+
+func protoToJSON(log *logging.Logger, message proto.Message) string {
+	result, err := jsonutil.ProtoToJSON(message, jsonutil.OptCompact, jsonutil.OptUnEscape)
+	if err != nil {
+		log.Error("Failed to convert proto to JSON: ", err)
+	} else {
+		return result
+	}
+	return ""
+}
+
+// LogSuccessfulUserLogin logs user attributes in the specified logger instance.
+func LogSuccessfulUserLogin(log *logging.Logger, user *v1.AuthStatus) {
+	serviceIdStr := ""
+	permissionsStr := ""
+	if user.GetServiceId() != nil {
+		serviceIdStr = protoToJSON(log, user.GetServiceId())
+	}
+	if user.GetUserInfo().GetPermissions() != nil {
+		permissionsStr = protoToJSON(log, user.GetServiceId())
```

Comment:
Shouldn't it be `user.GetUserInfo().GetPermissions()` that is passed to `protoToJSON` here ?